pennylane/_qubit_device.py

```diff
@@ -399,17 +399,17 @@ def estimate_probability(self, wires=None):
         Returns:
             List[float]: list of the probabilities
         """
-        wires = wires or self.register
+
+        wires = wires or self.wires
+        # convert to a wires object
         wires = Wires(wires)
-        # get indices of wires on the device's register
-        wire_indices = self.register.indices(wires)
-        # consider only the requested wires
-        wire_indices = np.hstack(wire_indices)
+        # translate to wire labels used by device
+        wires = self.map_wires(wires)
 
-        samples = self._samples[:, np.array(wire_indices)]
+        samples = self._samples[:, wires.toarray()]
```

Comment:
Added a few nice functions to the wire class :)

Reply:
🎉

Reply:
(Just a thought: if you want to define the `__array__` magic method on the `Wires` class, then you will be able to directly do `np.array(wires)`. You could simply have `__array__` return `self.toarray()`.

More details: https://numpy.org/devdocs/user/basics.dispatch.html)

Reply:
Nice idea!